src/utils/findProvidesModule.js

```diff
@@ -84,9 +84,11 @@ function findProvidesModule(directories, opts = {}) {
 
       // Throw when duplicated modules are provided from a different
       // fileName
+      /*
```

Comment:
Really the module map should be keeping track of (moduleName, platform) => file.  Since there is no rule that any foo.android.js and foo.ios.js file have to be in the same directory.  

But since platform isn't obviously available at the lookup time, I'm just doing last write wins here.  What this means practically is that you have to specify ['react-native', 'react-native-windows'] as you providesModuleNodeModules, so that the react-native-windows modules stomp on the react-native ones.  This also means that you have to only specify the react-native-windows package when building the windows platform, and not when building the ios/android platforms.